File: client/cs/src/Proton.Drive.Sdk/Telemetry/UploadPerformanceMetric.cs
namespace Proton.Drive.Sdk.Telemetry;

// Numeric values must match the UploadPerformanceMetric enum in proton.drive.sdk.proto.
public enum UploadPerformanceMetric
{
    Unspecified = 0,

    /// <summary>Throughput over active time for a file below the small size limit, in kibibytes per second.</summary>
    SmallFileThroughput = 1,

    /// <summary>Throughput over active time for a file at or above the small size limit, in kibibytes per second.</summary>
    LargeFileThroughput = 2,

    /// <summary>Share of total time spent running the pipeline, for uploads on the block route, in whole percent.</summary>
    LargeRouteActiveTimeShare = 3,

    /// <summary>Share of total time spent running the pipeline, for uploads on the small route, in whole percent.</summary>
    SmallRouteActiveTimeShare = 4,
}

File: incubating/client/kt/sdk/src/main/kotlin/me/proton/drive/sdk/telemetry/UploadPerformanceMetric.kt
package me.proton.drive.sdk.telemetry

enum class UploadPerformanceMetric {
    UNRECOGNIZED,
    UNSPECIFIED,
    SMALL_FILE_THROUGHPUT,
    LARGE_FILE_THROUGHPUT,
    LARGE_ROUTE_ACTIVE_TIME_SHARE,
    SMALL_ROUTE_ACTIVE_TIME_SHARE,
}
